packages/optimistic-oracle/src/proposer.js

```diff
@@ -175,8 +175,126 @@ class OptimisticOracleProposer {
     }
   }
 
+  // Submit disputes to proposed price requests
   async sendDisputes() {
-    // TODO
+    this.logger.debug({
+      at: "OptimisticOracleProposer#sendDisputes",
+      message: "Checking for undisputed price requests to dispute"
+    });
+
+    for (let priceRequest of this.optimisticOracleClient.getUndisputedProposals()) {
+      // Get proposal price
+      let proposalPrice = priceRequest.proposedPrice;
+
+      // Create pricefeed for identifier
+      const priceFeed = await this._createOrGetCachedPriceFeed(priceRequest.identifier);
+
+      // Pricefeed is either constructed correctly or is null.
+      if (!priceFeed) {
+        this.logger.error({
+          at: "OptimisticOracleProposer#sendDisputes",
+          message: "Failed to construct a PriceFeed for price request",
+          priceRequest
+        });
+        continue;
+      }
+
+      // With pricefeed successfully constructed, confirm the proposal price
+      await priceFeed.update();
+      let disputePrice;
+      try {
+        disputePrice = (await priceFeed.getHistoricalPrice(priceRequest.timestamp)).toString();
+      } catch (error) {
+        this.logger.error({
+          at: "OptimisticOracleProposer#sendDisputes",
+          message: "Failed to query historical price for price request",
+          disputePrice,
+          error
+        });
+        continue;
+      }
+
+      // If proposal price is not equal to the dispute price, then
+      // prepare dispute.
+      // TODO: Implement dispute buffer feature to only dispute if proposal and dispute price
```

Comment:
this is a great point and something we need to consider quite closely when it comes to funding rates. ideally a funding rate within a small margin of error cant be disputed as it'll become basically impossible to create a third party implementation that does not constantly get disputed due to small variants. There needs to be some window of tolerance for marginally wrong proposals that _dont_ get disputed. WDYT?

Reply:
I'll open up a subsequent PR on this I have one idea where we basically check precision up to X decimals of precision. For example, 1.2345 and 1.23456 are "equal" up to 4 decimals of precision